### PR TITLE
Meshing for `Annulus` primitive

### PR DESCRIPTION
**Pull request by mweatherley**
# Objective

Related to #10572 
Allow the `Annulus` primitive to be meshed.

## Solution

We introduce a `Meshable` structure, `AnnulusMeshBuilder`, which allows the `Annulus` primitive to be meshed, leaving optional configuration of the number of angular sudivisions to the user. Here is a picture of the annulus's UV-mapping:
Screenshot 2024-03-26 at 10 39 48 AM

Other features are essentially identical to the implementations for `Circle`/`Ellipse`. 

---

## Changelog

- Introduced `AnnulusMeshBuilder`
- Implemented `Meshable` for `Annulus` with `Output = AnnulusMeshBuilder`
- Implemented `From<Annulus>` and `From<AnnulusMeshBuilder>` for `Mesh`
- Added `impl_reflect!` declaration for `Annulus` and `Triangle3d` in `bevy_reflect`

---

### Design considerations

The only interesting wrinkle here is that the existing UV-mapping of `Ellipse` (and hence of `Circle` and `RegularPolygon`) is non-radial (it's skew-free, created by situating the mesh in a bounding rectangle), so the UV-mapping of `Annulus` doesn't limit to that of `Circle` as its inner radius tends to zero, for instance. I don't see this as a real issue for `Annulus`, which should almost certainly have this kind of UV-mapping, but I think we ought to at least consider allowing mesh configuration for `Circle`/`Ellipse` that performs radial UV-mapping instead. (In these cases in particular, it would be especially easy, since we wouldn't need a different parameter set in the builder.)

### CODE REVIEW

File: crates/bevy_render/src/mesh/primitives/dim2.rs [hunk 1]
```diff
@@ -193,6 +195,124 @@ impl From<EllipseMeshBuilder> for Mesh {
...
+        self
+    }
+
+    /// Builds a [`Mesh`] based on the configuration in `self`.
+    pub fn build(&self) -> Mesh {
+        let inner_radius = self.annulus.inner_circle.radius;
+        let outer_radius = self.annulus.outer_circle.radius;
+
+        let num_vertices = (self.resolution + 1) * 2;
+        let mut indices = Vec::with_capacity(num_vertices * 3);
```
Comment (Jondolf) [inline crates/bevy_render/src/mesh/primitives/dim2.rs:240]: I think this is slightly larger than necessary? The loop adds 2 sets of 3 indices `self.resolution` times, not `self.resolution + 1` times
```suggestion
        let mut indices = Vec::with_capacity(self.resolution * 6);
```

Comment (mweatherley) [reply crates/bevy_render/src/mesh/primitives/dim2.rs:240]: Nice catch!

**Review by Jondolf — APPROVED** [review]
CI is screaming but it seems to be for unrelated reasons (as usual). LGTM!

I agree it'd be nice to add radial/polar UV mapping as a configuration option for `Circle` and `Ellipse` (and `RegularPolygon`?) in a follow-up.

**Review by Chubercik — APPROVED** [review]
LGTM ^^

**Pull request merged**